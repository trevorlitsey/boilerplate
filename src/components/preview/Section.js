import React from 'react';
import { string } from 'prop-types';
import styled from 'styled-components';

const Container = styled.div`

	h4 {
		color: var(--secondary);
		text-decoration: underline;
	}
`

// needs to be class for refs!
class Section extends React.Component {

	static propTypes = {
		id: string.isRequired,
		title: string.isRequired,
		text: string.isRequired,
	}

	render() {

		const { id, title, text } = this.props;

		return (
			<Container>
				<h4 data-test="title" id={id}>{title}</h4>
				<p data-test="text" >{text}</p>
			</Container>
		)
	}
}

export default Section;