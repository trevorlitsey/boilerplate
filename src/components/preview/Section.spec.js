import React from 'react';
import { shallow } from 'enzyme';

import Section from './Section';

describe('Section', () => {

	it('should render without crashing', () => {
		const wrapper = renderSection();
		expect(wrapper).toMatchSnapshot();
	})

	it('should render correct title and text', () => {

		const title = 'WHAT A GREAT TITLE';
		const text = 'WHAT A GREAT TEXT';

		const wrapper = renderSection({ title, text });

		expect(wrapper.find('[data-test="title"]').text()).toBe(title);
		expect(wrapper.find('[data-test="text"]').text()).toBe(text);

	})

})


// -------
function renderSection(props = {}) {
	const propsToUser = {
		id: '8779y2hh',
		title: 'this is a title!',
		text: 'this is some text!',
		...props,
	}
	return shallow(<Section {...propsToUser} />)
}