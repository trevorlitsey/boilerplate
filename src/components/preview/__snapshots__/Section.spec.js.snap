// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Section
    id="8779y2hh"
    text="this is some text!"
    title="this is a title!"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <h4
          data-test="title"
          id="8779y2hh"
        >
          this is a title!
        </h4>,
        <p
          data-test="text"
        >
          this is some text!
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "this is a title!",
          "data-test": "title",
          "id": "8779y2hh",
        },
        "ref": null,
        "rendered": "this is a title!",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "this is some text!",
          "data-test": "text",
        },
        "ref": null,
        "rendered": "this is some text!",
        "type": "p",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <h4
            data-test="title"
            id="8779y2hh"
          >
            this is a title!
          </h4>,
          <p
            data-test="text"
          >
            this is some text!
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "this is a title!",
            "data-test": "title",
            "id": "8779y2hh",
          },
          "ref": null,
          "rendered": "this is a title!",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "this is some text!",
            "data-test": "text",
          },
          "ref": null,
          "rendered": "this is some text!",
          "type": "p",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
